packages/backend/src/open_payments/quote/service.ts

```diff
@@ -184,21 +193,48 @@ async function createQuote(
         description: 'Time to getQuote'
       }
     )
-    const quote = await deps.paymentMethodHandlerService.getQuote(
-      paymentMethod,
-      {
+    let quote: PaymentQuote
+    try {
+      quote = await deps.paymentMethodHandlerService.getQuote(paymentMethod, {
         quoteId,
         walletAddress,
         receiver,
         receiveAmount: options.receiveAmount,
         debitAmount: options.debitAmount
           ? {
               ...options.debitAmount,
-              value: options.debitAmount.value - fixedDebitFee
+              value: quoteDebitAmount
             }
           : undefined
+      })
+    } catch (err) {
+      if (
+        err instanceof PaymentMethodHandlerError &&
+        err.code === PaymentMethodHandlerErrorCode.QuoteNonPositiveReceiveAmount
+      ) {
+        let details = undefined
+
+        if (err.details?.minSendAmount) {
+          details = {
+            minSendAmount: {
+              value: 0n,
+              assetCode: walletAddress.asset.code,
+              assetScale: walletAddress.asset.scale
+            }
+          }
+          const quoteMinSendAmount = err.details.minSendAmount as bigint
+          const fixedFee = sendingFee?.fixedFee ?? 0n
+
+          details.minSendAmount.value =
+            quoteMinSendAmount +
+            (sendingFee?.calculate(fixedFee + quoteMinSendAmount) ?? 0n)
```

Comment:
Since `calculate()` already adds the `fixedFee` in the method
```suggestion
            (sendingFee?.calculate(quoteMinSendAmount) ?? 0n)
```